website/docs/core-concepts/connecting-to-data-sources/authentication/README.md

```diff
@@ -1,63 +1,120 @@
+
+In the context of a REST API, one-off queries are typically used when someone needs a quick answer to a specific question or problem, rather than engaging in a longer, more involved process. They're often used to retrieve or manipulate data from a server in real-time.
+
+
+ <VideoEmbed host="youtube" videoId="IptCmvKdbog" title="Add new API" caption="Add new REST API"/> 
+
+You can provide the URL and additional information like the Headers, Params, Body, and Pagination. Once you have added code in the body section, you can directly run the query and connect it with widgets. To learn more about configuring APIs, you can refer to the documentation on [configuring APIs](/core-concepts/connecting-to-data-sources/authentication/connect-to-apis).
+
+
+### Connecting to a authenticated API
+
+A one-off query is a single request for information or assistance, while an authenticated API is an API that requires authentication before allowing access to its resources or capability. This is done to ensure that only authorized users are able to interact with the API and to protect sensitive data or capability from unauthorized access:
 
  <VideoEmbed host="youtube" videoId="Uy7ZDviGbtM" title="Add new API" caption="Add new API"/> 
 
 
-You’ll be redirected to the create datasource page as shown below:
+* Click on the **+** icon next to the Datasources and choose **Authenticated API**.
+* Provide the configuration details required to connect to your API.
+* Rename and save your datasource.
+
+
+
+Once your datasource has been added successfully, a success pop-up appears at the top. 
+
+
+### Importing CURL commands
+
+CURL is a command-line tool that can be used to make HTTP requests to a server. You can use CURL to send various types of HTTP requests, such as GET, POST, PUT, and DELETE, to a server to retrieve or manipulate data. Appsmith makes it easy to import your APIs into your application using CURL commands.
+
+
+To import CURL commands into your application:
 
-![Create a new Datasource](</img/OAuth__API_Integration__Create_New_DB.png>)
+* Click on the **+** icon next to the Datasources and choose **CURL Import**.
+* Add your command, for example:
 
-## Configure
+```js
+curl -X GET https://example.com/resource
+```
 
-You can configure the data source as follows:
+![](</img/import_curl_(1).gif>)
 
-### Name
+## Passing data/parameters to API calls
 
-By default, Appsmith supplies a running sequence for the data source created. You can click on the pencil icon next to it to edit the name of the data source.
+There are various ways to pass parameters to API calls on Appsmith. This video shows three ways to pass data to API calls:
 
-:::info
-It is advisable that you give a meaningful name to your Authenticated API datasource.
-:::
+<VideoEmbed host="youtube" videoId="znaaDiQbAS8" title="How to pass parameters to an API call" caption="How to pass parameters to an API call"/>
 
-### URL
+## Mock API
```

Comment:
This looks out of place here. As it's part of REST API, can we add it to the REST API section?